src/libraries/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -72,11 +72,11 @@ internal static bool IsValidDriveChar(char value)
             return (value >= 'A' && value <= 'Z') || (value >= 'a' && value <= 'z');
         }
 
-        internal static bool EndsWithPeriodOrSpace(string? path)
-        {
-            if (string.IsNullOrEmpty(path))
-                return false;
+        internal static bool EndsWithPeriodOrSpace(string? path) =>
+            !string.IsNullOrEmpty(path) && EndsWithPeriodOrSpaceSlim(path);
 
+        internal static bool EndsWithPeriodOrSpaceSlim(string path)
```

Comment:
Looks like this method can be turned into a single line and produce the same code:
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHufuBLAHYYYUAdgA2TckgbAIESQEEAFMXIAGBgAdsGABYBKLrx6dqJi0wDs23XoDaO/QDoAMjAEBzfQzgNyALoMALzBDADkEQyEhLb6jnZuHt56vv5BoRHO4QDcxhYAvvkMxbyCwqISUjJyCgwAQqoacYalpm0WYHrYUAxgIS0JLu5ePn6BeeaWvMQ2/ZmRkTF9IWHh2ZPTRdQFQA===
(checked both 32 and 64 bit)

Given that, I suggest inlining `path[path.Length - 1] == ' ' || path[path.Length - 1] == '.'` into EndsWithPeriodOrSpace and EnsureExtendedPrefixIfNeeded. This eliminates a method that isn't really useful, but also is a trap for anyone in future that calls it without checking the length. It's perfectly clear when inlined.

Reply:
BTW, we have slightly changed the behavior here. Previously EndsWithPeriodOrSpace would return false for "      " and now it will return true. Is that OK? I'm actually not sure.

If it is, then EndsWithPeriodOrSpace could be even less code: `return path.Length > 0 && (path[path.Length - 1] == ' ' || path[path.Length - 1] == '.');` generates less code. But, I don't suggest changing that here.

Reply:
We have only two direct usages, which could break:

### PathInternal.EnsureExtendedPrefixIfNeeded
```cs
        [return: NotNullIfNotNull("path")]
        internal static string? EnsureExtendedPrefixIfNeeded(string? path)
        {
            if (path != null && (path.Length >= MaxShortPath || EndsWithPeriodOrSpaceSlim(path)))
            {
                return EnsureExtendedPrefix(path);
            }
            else
            {
                return path;
            }
        }
    ```

and

### FileSystemInfo.NormalizedPath
```cs
        internal string NormalizedPath
            => PathInternal.EndsWithPeriodOrSpace(FullPath) ? PathInternal.EnsureExtendedPrefix(FullPath) : FullPath;
```

@tmds @Jozkee Are you comfortable with this change (Tests are passing, but behavior might slightly change)?

Reply:
@deeprobin I'd like to err on the side of caution here and avoid the subtle behavior change.